### PR TITLE
#1569 Issue - adopt best practices Stripe Source

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*Adopt connector Best Practices for Stripe Source.*

## Checklist
- [x] Check connection verifies permissions upfront
- [x] Check connection failure emits actionable error messages. This will probably require you to think about the failure modes that can occur (e.g: timeout, permission not granted, token does not exist, etc.. -- make sure that you add custom integration tests to verify each of these failure modes, or unit tests if you can mock the HTTP requests involved)
- [x] If API connector: integration tests validate that every stream outputs data (you may need to seed the underlying API with fake data). This will establish a baseline confidence that the connector works correctly on every data stream.
- [x] If API connector: backs off on hitting rate limits (instead of just failing/exiting).
- [x] If API connector: implements incremental sync (where possible -- not all streams will allow this)

### CODE REVIEW

**Comment by yevhenii-ldv** [comment]
I filled Stripe account with fake data using the API for the following streams:
```
charges
events
customers
plans
invoices
invoices_item
products
invoice_line_items
coupons
```

I could not add fake data to some streams, this is directly due to the fact that all these streams are based on bank transaction. I didn't possible to carry out these operations through the API even with test cards.:
```
transfers
subscriptions
subscriptions_items
payouts
payouts_transactions
disputes
```

**Comment by yevhenii-ldv** [comment]
Rollback for this source has not been implemented yet. Singer Tap does not support rate limit handling.
I suppose we have to fork Stripe Tap and add handling for rate limits.

**Comment by yevhenii-ldv** [comment]
A small suggestion, in the config for this source, change the start date from 2017 to 2021 or at least mid-2020, because during integration tests, streams with an incremental are processed for a very long time (more than 30 minutes).

**Comment by yevhenii-ldv** [comment]
Updated Source for backoff all requests

File: airbyte-integrations/connectors/source-stripe-singer/setup.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
     packages=find_packages(),
     package_data={"": ["*.json"]},
     install_requires=[
-        "tap-stripe==1.4.4",
+        "tap-stripe @ https://github.com/yevhenii-ldv/tap-stripe/tarball/master#egg=tap-stripe-1.4.5",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-stripe-singer/setup.py:35]: can you use github.com/airbytehq/tap-stripe ? it should be the same as your fork.

**Review by sherifnada — COMMENTED** [review]
@yevhenii-ldv 
1. I changed the start date as you suggested and updated it in lastpass. please take a look for the updated config
2. Let's discuss seeding with fake data in tomorrow's conversation

File: airbyte-integrations/connectors/source-stripe-singer/setup.py (hunk 1 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-stripe-singer/setup.py]: updated

**Comment by yevhenii-ldv** [comment]
I created PR for update tap-stripe: https://github.com/airbytehq/tap-stripe/pull/1

**Comment by sherifnada** [comment]
@yevhenii-ldv I added data to the following streams from the Stripe UI/API:

* **Transfers** -- had to be done manually
* **Subscriptions** already had data
* **subscriptions_items** already had data
* **payouts** already had data
* **payouts_transactions** already had data (the API endpoint is called balance_transactions if you want to verify) 
* **disputes** I've added a disputed payment via the Ui

**Pull request merged**